Explorer/Assets/DCL/Input/Systems/UpdateCursorInputSystem.cs

```diff
@@ -0,0 +1,312 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.Character.CharacterCamera.Components;
+using DCL.CharacterCamera;
+using DCL.CharacterCamera.Components;
+using DCL.CharacterCamera.Systems;
+using DCL.Diagnostics;
+using DCL.Input.Crosshair;
+using DCL.Interaction.PlayerOriginated.Components;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+using UnityEngine.UIElements;
+using Button = UnityEngine.UIElements.Button;
+using Toggle = UnityEngine.UIElements.Toggle;
+
+namespace DCL.Input.Systems
+{
+    [UpdateInGroup(typeof(InputGroup))]
+    [UpdateBefore(typeof(UpdateCameraInputSystem))]
+    [LogCategory(ReportCategory.INPUT)]
+    public partial class UpdateCursorInputSystem : UpdateInputSystem<CameraInput, CameraComponent>
+    {
+        private const int MOUSE_BOUNDS_OFFSET = 10;
+        private static readonly Vector2 CURSOR_OFFSET = new (0, 15);
+
+        private readonly IEventSystem eventSystem;
+        private readonly ICursor cursor;
+        private readonly ICrosshairView crosshairCanvas;
+        private readonly DCLInput.CameraActions cameraActions;
+        private readonly DCLInput.UIActions uiActions;
+        private bool hasHoverCollider;
+        private bool isAtDistance;
+        private bool isHoveringAnInteractable;
+        private bool wantsToUnlockForced;
+        private readonly Dictionary<GameObject, bool> interactionCache = new ();
```

Comment:
Is it possible to encapsulate caches with some functionality in separate classes as I believe it's too much for a single unit?